src/libsyntax_pos/hygiene.rs

```diff
@@ -174,7 +174,6 @@ crate struct HygieneData {
     marks: Vec<MarkData>,
     syntax_contexts: Vec<SyntaxContextData>,
     markings: FxHashMap<(SyntaxContext, Mark, Transparency), SyntaxContext>,
-    default_edition: Edition,
```

Comment:
Looking a bit into the future, `expn_info` in `MarkData` should be mandatory rather than optional, it will mean that the "default edition" will be available as `marks[0].expn_info.edition` and set from `syntax_pos::Globals::new`.
In this sense I don't see much reason to move it from here to some other location.

Reducing reliance on the global data and using `span_sess.edition` is very much reasonable though.
After addressing my comments `Span::edition` will be pretty much the only place using the global edition, which cannot be removed easily.

Reply:
I pretty much only moved it to avoid the lock when accessing it.